<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Zeya Wang and Fan Gao" />

<meta name="date" content="2019-02-15" />

<title>A Vignette for DeMixT</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">A Vignette for DeMixT</h1>
<h4 class="author"><em>Zeya Wang and Fan Gao</em></h4>
<h4 class="date"><em>2019-02-15</em></h4>


<div id="TOC">
true
</div>

<div id="introduction" class="section level1">
<h1>1. Introduction</h1>
<p>Transcriptomic deconvolution in cancer and other heterogeneous tissues remains challenging. Available methods lack the ability to estimate both component-specific proportions and expression profiles for individual samples. We develop a three-component deconvolution model, DeMixT, for expression data from a mixture of cancerous tissues, infiltrating immune cells and tumor microenvironment. DeMixT is a software package that performs deconvolution on transcriptome data from a mixture of two or three components.</p>
<p>DeMixT is a frequentist-based method and fast in yielding accurate estimates of cell proportions and compart-ment-specific expression profiles for two-component and three-component deconvolution problem. Our method promises to provide deeper insight into cancer biomarkers and assist in the development of novel prognostic markers and therapeutic strategies.</p>
<p>The function DeMixT is designed to finish the whole pipeline of deconvolution for two or three components. DeMixT.S1 function is designed to estimate the proportions of all mixed samples for each mixing component. DeMixT.S2 function is designed to estimate the component-specific deconvolved expressions of individual mixed samples for a given set of genes.</p>
</div>
<div id="installation" class="section level1">
<h1>2. Installation</h1>
<div id="source-file" class="section level2">
<h2>2.1 Source file</h2>
<p>DeMixT source files are compatible with Windows, Linux and macOS.</p>
<p>DeMixT_0.2 is the latest version, which is for a computer that has OpenMP. DeMixT_0.2.1 is for computers that do not have OpenMP. To install DeMixT_0.2.1, start R and enter:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#devtools::install_github(&quot;wwylab/DeMixTallmaterials/DeMixT_0.2.1&quot;)</span></code></pre></div>
</div>
<div id="more-information" class="section level2">
<h2>2.2 More information</h2>
<p>For a computer that does not have OpenMP or the older versions of DeMixT, please visit <a href="https://github.com/wwylab/DeMixTallmaterials" class="uri">https://github.com/wwylab/DeMixTallmaterials</a>. You can also download the installation files directly from the website:</p>
<p>DeMixT_0.2: <a href="http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.tar.gz" class="uri">http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.tar.gz</a></p>
<p>DeMixT_0.2.1: <a href="http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.1.tar.gz" class="uri">http://bioinformatics.mdanderson.org/Software/DeMixT/DeMixT_0.2.1.tar.gz</a></p>
<p>For more information, please visit: <a href="http://bioinformatics.mdanderson.org/main/DeMixT" class="uri">http://bioinformatics.mdanderson.org/main/DeMixT</a></p>
</div>
<div id="functions" class="section level2">
<h2>2.3 Functions</h2>
<p>The following table shows the functions included in DeMixT.</p>
<table style="width:68%;">
<colgroup>
<col width="19%"></col>
<col width="48%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Table Header</th>
<th>Second Header</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>DeMixT</td>
<td>Deconvolution of heterogeneous tumor samples with two or three components using expression data from RNAseq or microarray platforms</td>
</tr>
<tr class="even">
<td>DeMixT.S1</td>
<td>Estimates the proportions of mixed samples for each mixing component</td>
</tr>
<tr class="odd">
<td>DeMixT.S2</td>
<td>Deconvolves expressions of each individual sample for unknown component</td>
</tr>
<tr class="even">
<td>Optimum.KernelC</td>
<td>Kernel function to call the C function used for parameter estimation in DeMixT</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-info" class="section level1">
<h1>3. Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>(<span class="dt">package =</span> <span class="st">&quot;DeMixT&quot;</span>)</code></pre></div>
<pre><code>## R version 3.5.2 (2018-12-20)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.3
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## character(0)
## 
## other attached packages:
## [1] DeMixT_0.3.1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.18      rstudioapi_0.7    knitr_1.20       
##  [4] magrittr_1.5      usethis_1.4.0     devtools_2.0.1   
##  [7] grDevices_3.5.2   pkgload_1.0.1     R6_2.2.2         
## [10] rlang_0.3.0.1     stringr_1.3.1     tools_3.5.2      
## [13] utils_3.5.2       pkgbuild_1.0.2    sessioninfo_1.1.0
## [16] cli_1.0.0         withr_2.1.2       htmltools_0.3.6  
## [19] stats_3.5.2       remotes_2.0.1     datasets_3.5.2   
## [22] yaml_2.2.0        assertthat_0.2.0  digest_0.6.16    
## [25] rprojroot_1.3-2   base_3.5.2        crayon_1.3.4     
## [28] processx_3.2.0    callr_3.0.0       base64enc_0.1-3  
## [31] graphics_3.5.2    fs_1.2.6          ps_1.1.0         
## [34] evaluate_0.11     memoise_1.1.0     glue_1.3.0       
## [37] rmarkdown_1.10    stringi_1.2.4     compiler_3.5.2   
## [40] methods_3.5.2     desc_1.2.0        backports_1.1.2  
## [43] prettyunits_1.0.2</code></pre>
</div>
<div id="methods" class="section level1">
<h1>4. Methods</h1>
<div id="model" class="section level2">
<h2>4.1 Model</h2>
<p>Let <span class="math inline">\(Y_{ig}\)</span> be the observed expression levels of the raw measured data from clinically derived malignant tumor samples for gene <span class="math inline">\(g, g = 1, \cdots, G\)</span> and sample <span class="math inline">\(i, i = 1, \cdots, S\)</span>. <span class="math inline">\(G\)</span> denotes the total number of probes/genes and <span class="math inline">\(S\)</span> denotes the number of samples. The observed expression levels for solid tumors can be modeled as a linear combination of raw expression levels from three components: <span class="math display">\[ {Y_{ig}} = \pi _{1,i}N_{1,ig} + \pi _{2,i}N_{2,ig} + (1 - \pi_{1,i} - \pi _{2,i}){T_{ig}} \label{eq:1} \]</span></p>
<p>Here <span class="math inline">\(N_{1,ig}\)</span>, <span class="math inline">\(N_{2,ig}\)</span> and <span class="math inline">\({T_{ig}}\)</span> are the unobserved raw expression levels from each of the three components. We call the two components for which we require reference samples the <span class="math inline">\(N_1\)</span>-component and the <span class="math inline">\(N_2\)</span>-component. We call the unknown component the T-component. We let <span class="math inline">\(\pi_{1,i}\)</span> denote the proportion of the <span class="math inline">\(N_1\)</span>-component, <span class="math inline">\(\pi_{2,i}\)</span> denote the proportion of the <span class="math inline">\(N_2\)</span>-component, and <span class="math inline">\(1 - \pi_{1,i} - \pi_{2,i}\)</span> denote the proportion of the T-component. We assume that the mixing proportions of one specific sample remain the same across all genes.</p>
<p>Our model allows for one component to be unknown, and therefore does not require reference profiles from all components. A set of samples for <span class="math inline">\(N_{1,ig}\)</span> and <span class="math inline">\(N_{2,ig}\)</span>, respectively, needs to be provided as input data. This three-component deconvolution model is applicable to the linear combination of any three components in any type of material. It can also be simplified to a two-component model, assuming there is just one <span class="math inline">\(N\)</span>-component. For application in this paper, we consider tumor (<span class="math inline">\(T\)</span>), stromal (<span class="math inline">\(N_1\)</span>) and immune components (<span class="math inline">\(N_2\)</span>) in an admixed sample (<span class="math inline">\(Y\)</span>).</p>
<p>Following the convention that <span class="math inline">\(\log_2\)</span>-transformed microarray gene expression data follow a normal distribution, we assume that the raw measures <span class="math inline">\(N_{1,ig} \sim LN({\mu _{{N_1}g}},\sigma _{{N_1}g}^2)\)</span>, <span class="math inline">\(N_{2,ig} \sim LN({\mu _{{N_2}g}},\sigma _{{N_2}g}^2)\)</span> and <span class="math inline">\({T_{ig}} \sim LN({\mu _{Tg}}, \sigma _{Tg}^2)\)</span>, where LN denotes a <span class="math inline">\(\log_2\)</span>-normal distribution and <span class="math inline">\(\sigma _{{N_1}g}^2\)</span>,<span class="math inline">\(\sigma _{{N_2}g}^2\)</span>,<span class="math inline">\(\sigma _{Tg}^2\)</span> reflect the variations under <span class="math inline">\(\log_2\)</span>-transformed data. Consequently, our model can be expressed as the convolution of the density function for three <span class="math inline">\(\log_2\)</span>-normal distributions. Because there is no closed form of this convolution, we use numerical integration to evaluate the complete likelihood function (see the full likelihood in the Supplementary Materials).</p>
</div>
<div id="the-demixt-algorithm-for-deconvolution" class="section level2">
<h2>4.2 The DeMixT algorithm for deconvolution</h2>
<p>DeMixT estimates all distribution parameters and cellular proportions and reconstitutes the expression profiles for all three components for each gene and each sample. The estimation procedure (summarized in Figure 1b) has two main steps as follows.</p>
<ol style="list-style-type: decimal">
<li><p>Obtain a set of parameters <span class="math inline">\(\{\pi_{1,i}, \pi_{2,i}\}_{i=1}^S\)</span>, <span class="math inline">\(\{\mu_T, \sigma_T\}_{g=1}^G\)</span> to maximize the complete likelihood function, for which <span class="math inline">\(\{\mu_{N_{1,g}}, \sigma_{N_{1,g}}, \mu_{N_{2,g}}, \sigma_{N_{2,g}}\}_{g=1}^G\)</span> were already estimated from the available unmatched samples of the <span class="math inline">\(N_1\)</span> and <span class="math inline">\(N_2\)</span> component tissues. (See further details in our paper.)</p></li>
<li><p>Reconstitute the expression profiles by searching each set of <span class="math inline">\(\{n_{1,ig},n_{2,ig}\}\)</span> that maximizes the joint density of <span class="math inline">\(N_{1,ig}\)</span>, <span class="math inline">\(N_{2,ig}\)</span> and <span class="math inline">\(T_{ig}\)</span>. The value of <span class="math inline">\(t_{ig}\)</span> is solved as <span class="math inline">\({y_{ig}} - {{\hat \pi }_{1,i}}{n_{1,ig}} - {{\hat \pi }_{2,i}}{n_{2,ig}}\)</span>.</p></li>
</ol>
<p>These two steps can be separately implemented using the function DeMixT.S1 and DeMixT.S2, which are combined in the function DeMixT.</p>
</div>
</div>
<div id="examples" class="section level1">
<h1>5. Examples</h1>
<div id="simulated-two-component-data" class="section level2">
<h2>5.1 Simulated two-component data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DeMixT)</code></pre></div>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colMeans, colnames, colSums, dirname, do.call, duplicated,
##     eval, evalq, Filter, Find, get, grep, grepl, intersect,
##     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
##     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
##     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
##     table, tapply, union, unique, unsplit, which, which.max,
##     which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation(&quot;Biobase&quot;)', and for packages 'citation(&quot;pkgname&quot;)'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     aperm, apply</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(test.data.y)</code></pre></div>
<pre><code>## Warning in data(test.data.y): data set 'test.data.y' not found</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(test.data.comp1)</code></pre></div>
<pre><code>## Warning in data(test.data.comp1): data set 'test.data.comp1' not found</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res &lt;-<span class="st"> </span><span class="kw">DeMixT</span>(<span class="dt">data.Y =</span> test.data1.y, <span class="dt">data.comp1 =</span> test.data1.comp1, <span class="dt">if.filter =</span> <span class="ot">FALSE</span>, <span class="dt">output.more.info =</span> <span class="ot">TRUE</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res<span class="op">$</span>pi</code></pre></div>
<pre><code>##             1         2         3         4         5         6         7
## pi1 0.1074078 0.2196809 0.3047277 0.3523041 0.4248352 0.4888389 0.6179312
##            8         9        10
## pi1 0.608743 0.6632821 0.7670111</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(res<span class="op">$</span>ExprT, <span class="dv">3</span>)</code></pre></div>
<pre><code>##          1         2         3        4        5        6        7
## 1 96.83726  83.56023  80.82836 80.72725 84.47915 83.32202 87.44895
## 2 77.09617 112.44931 136.70835 85.97539 81.43849 68.38339 87.89490
## 3 38.68176  35.21445  38.30982 37.65722 32.67902 73.87686 83.91727
##           8        9       10
## 1  87.23697 85.60188 85.84117
## 2 101.27924 98.38278 72.67417
## 3  47.83651 47.25389 56.85126</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(res<span class="op">$</span>ExprN1, <span class="dv">3</span>)</code></pre></div>
<pre><code>##           1         2         3         4         5         6         7
## 1 129.09181 115.73951 105.97676 103.13745 116.22402 108.05379 172.66879
## 2  90.39363  96.06401 102.69785  92.02293  89.36471  77.15229  96.40235
## 3  53.11342  52.57825  52.81754  52.58788  50.68354  57.91523  62.84381
##           8         9        10
## 1 162.01438 130.74263 153.04748
## 2 116.43629 120.28931  65.71293
## 3  56.23678  56.85083  68.33056</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(res<span class="op">$</span>Mu, <span class="dv">3</span>)</code></pre></div>
<pre><code>##       MuN1      MuT
## 1 6.940477 6.420368
## 2 6.582707 6.493136
## 3 5.767947 5.531644</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(res<span class="op">$</span>Sigma, <span class="dv">3</span>)</code></pre></div>
<pre><code>##     SigmaN1    SigmaT
## 1 0.2675802 0.1339120
## 2 0.3201497 0.3714339
## 3 0.2231435 0.5109864</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res<span class="op">$</span>pi.iter</code></pre></div>
<pre><code>## , , 1
## 
##             [,1]      [,2]      [,3]      [,4]      [,5]      [,6]
##  [1,] 0.09043013 0.2139893 0.2937994 0.3356124 0.4489045 0.5187335
##  [2,] 0.09198396 0.2256721 0.3003950 0.3433463 0.4443040 0.5215801
##  [3,] 0.09218502 0.2259228 0.3000308 0.3437418 0.4438475 0.5214382
##  [4,] 0.10617830 0.2252988 0.3015530 0.3608652 0.4454556 0.5067351
##  [5,] 0.10977845 0.2275840 0.3089533 0.3611274 0.4456627 0.5040962
##  [6,] 0.11123518 0.2424661 0.3081845 0.3580924 0.4228940 0.5039239
##  [7,] 0.11172841 0.2424911 0.3064393 0.3525787 0.4233200 0.4874916
##  [8,] 0.11071479 0.2216530 0.3054770 0.3513610 0.4243426 0.4873702
##  [9,] 0.10976194 0.2201011 0.3011756 0.3494824 0.4265594 0.4883885
## [10,] 0.10864671 0.2200628 0.3032327 0.3513427 0.4246896 0.4886372
##            [,7]      [,8]      [,9]     [,10]
##  [1,] 0.6544423 0.7281499 0.8230019 0.8744931
##  [2,] 0.6532641 0.7273786 0.8232727 0.8754047
##  [3,] 0.6528566 0.7269844 0.8231059 0.8336147
##  [4,] 0.6526147 0.7277983 0.7440691 0.8335681
##  [5,] 0.5813924 0.6121490 0.7436529 0.7410122
##  [6,] 0.5811898 0.6119788 0.7448583 0.7415068
##  [7,] 0.5804985 0.6122466 0.7454209 0.7417283
##  [8,] 0.5806471 0.6130025 0.7456191 0.7651868
##  [9,] 0.6154044 0.6110913 0.7453916 0.7661025
## [10,] 0.6176156 0.6089009 0.7454029 0.7668798</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res<span class="op">$</span>gene.name</code></pre></div>
<pre><code>##   [1] &quot;1&quot;   &quot;2&quot;   &quot;3&quot;   &quot;4&quot;   &quot;5&quot;   &quot;6&quot;   &quot;7&quot;   &quot;8&quot;   &quot;9&quot;   &quot;10&quot;  &quot;11&quot; 
##  [12] &quot;12&quot;  &quot;13&quot;  &quot;14&quot;  &quot;15&quot;  &quot;16&quot;  &quot;17&quot;  &quot;18&quot;  &quot;19&quot;  &quot;20&quot;  &quot;21&quot;  &quot;22&quot; 
##  [23] &quot;23&quot;  &quot;24&quot;  &quot;25&quot;  &quot;26&quot;  &quot;27&quot;  &quot;28&quot;  &quot;29&quot;  &quot;30&quot;  &quot;31&quot;  &quot;32&quot;  &quot;33&quot; 
##  [34] &quot;34&quot;  &quot;35&quot;  &quot;36&quot;  &quot;37&quot;  &quot;38&quot;  &quot;39&quot;  &quot;40&quot;  &quot;41&quot;  &quot;42&quot;  &quot;43&quot;  &quot;44&quot; 
##  [45] &quot;45&quot;  &quot;46&quot;  &quot;47&quot;  &quot;48&quot;  &quot;49&quot;  &quot;50&quot;  &quot;51&quot;  &quot;52&quot;  &quot;53&quot;  &quot;54&quot;  &quot;55&quot; 
##  [56] &quot;56&quot;  &quot;57&quot;  &quot;58&quot;  &quot;59&quot;  &quot;60&quot;  &quot;61&quot;  &quot;62&quot;  &quot;63&quot;  &quot;64&quot;  &quot;65&quot;  &quot;66&quot; 
##  [67] &quot;67&quot;  &quot;68&quot;  &quot;69&quot;  &quot;70&quot;  &quot;71&quot;  &quot;72&quot;  &quot;73&quot;  &quot;74&quot;  &quot;75&quot;  &quot;76&quot;  &quot;77&quot; 
##  [78] &quot;78&quot;  &quot;79&quot;  &quot;80&quot;  &quot;81&quot;  &quot;82&quot;  &quot;83&quot;  &quot;84&quot;  &quot;85&quot;  &quot;86&quot;  &quot;87&quot;  &quot;88&quot; 
##  [89] &quot;89&quot;  &quot;90&quot;  &quot;91&quot;  &quot;92&quot;  &quot;93&quot;  &quot;94&quot;  &quot;95&quot;  &quot;96&quot;  &quot;97&quot;  &quot;98&quot;  &quot;99&quot; 
## [100] &quot;100&quot; &quot;101&quot; &quot;102&quot; &quot;103&quot; &quot;104&quot; &quot;105&quot; &quot;106&quot; &quot;107&quot; &quot;108&quot; &quot;109&quot; &quot;110&quot;
## [111] &quot;111&quot; &quot;112&quot; &quot;113&quot; &quot;114&quot; &quot;115&quot; &quot;116&quot; &quot;117&quot; &quot;118&quot; &quot;119&quot; &quot;120&quot;</code></pre>
</div>
<div id="simulated-three-component-data" class="section level2">
<h2>5.2 Simulated three-component data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># data(test.data2.y)</span>
<span class="co"># data(test.data2.comp1)</span>
<span class="co"># data(test.data2.comp2)</span>
<span class="co"># res &lt;- DeMixT(data.Y = test.data2.y, data.comp1 = test.data2.comp1, data.comp2 = test.data2.comp2, if.filter = FALSE)</span></code></pre></div>
</div>
<div id="laser-capture-microdissection-lcm-prostate-cancer-ffpe-microarray-dataset" class="section level2">
<h2>5.3 Laser-capture microdissection (LCM) prostate cancer FFPE microarray dataset</h2>
<p>This dataset was generated at the Dana Farber Cancer Institute (GSE97284). Radical prostatectomy specimens were annotated in detail by pathologists, and regions of interest were identified that corresponded to benign epithelium, prostatic intraepithelial neoplasia (abnormal tissue that is possibly precancerous), and tumor, each with its surrounding stroma. FFPE samples are known to generate overall lower quality expression data than those from fresh frozen samples. We observed a small proportion of probesets that presented large differences in mean expression levels between the dissected tissues: tumor (<span class="math inline">\(T\)</span>) and stroma (<span class="math inline">\(N\)</span>) in this dataset. Only 53 probesets presented a mean difference <span class="math inline">\((| \bar{T} - \bar{N} |) &gt;1\)</span>, as compared to 10,397 probesets in GSE19830. We therefore chose the top 80 genes with the largest mean differences and ran DeMixT under two settings: tumor unknown and stroma unknown. DeMixT is able to obtain concordant estimates of the tumor proportions when the proportion of the stromal component was unknown and when the proportion of tumor tissue was unknown and also tended to provide accurate component-specific mean expression levels.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># library(DeMixT)</span>
<span class="co"># data &lt;- as.matrix(read.table(&quot;Data/input.lcm.txt&quot;, header = FALSE))</span>
<span class="co"># normal &lt;- data[, 1:25]</span>
<span class="co"># adm &lt;- data[, 26:48]</span>
<span class="co"># tumor &lt;- data[, 49:73]</span>
<span class="co"># </span>
<span class="co"># testr.TA &lt;- DeMixT(data.Y = 2^adm, data.comp1 = 2^tumor, </span>
<span class="co">#                    niter = 20, nbin = 60, if.filter = FALSE, tol = 10^-6)</span>
<span class="co"># testr.SA &lt;- DeMixT(data.Y = 2^adm, data.comp1 = 2^normal, </span>
<span class="co">#                    niter = 20, nbin = 60, if.filter = FALSE, tol = 10^-6)</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># # plot A</span>
<span class="co"># dt_purT &lt;- 1- as.numeric(testr.SA$pi)</span>
<span class="co"># dt_purS &lt;- 1- as.numeric(testr.TA$pi)</span>
<span class="co"># plot(1 - dt_purS, dt_purT, </span>
<span class="co">#      col = &quot;blue&quot;, pch = 1, xlim = c(0, 1), ylim = c(0, 1), </span>
<span class="co">#      xlab = expression(1 - hat(pi)[S]), ylab = expression(hat(pi)[T]))</span>
<span class="co"># abline(0, 1, col = &quot;red&quot;, lwd = 2)</span>
<span class="co"># </span>
<span class="co"># # Plot B - Mean expressions for Tumor</span>
<span class="co"># OB_St &lt;- log2(read.table(&quot;Data/lcm_normal.txt&quot;, header = F))</span>
<span class="co"># OB_Tu &lt;- log2(read.table(&quot;Data/lcm_tumor.txt&quot;, header = F))</span>
<span class="co"># DT_Tu_mu &lt;- as.numeric(testr.SA$Mu[, 1])</span>
<span class="co"># DT_St_mu &lt;- as.numeric(testr.TA$Mu[, 1])</span>
<span class="co"># DT_Tu_sg &lt;- as.numeric(testr.SA$Sigma[, 1])</span>
<span class="co"># DT_St_sg &lt;- as.numeric(testr.TA$Sigma[, 1])</span>
<span class="co"># OB_St_m &lt;- apply(OB_St, 1, mean)</span>
<span class="co"># OB_Tu_m &lt;- apply(OB_Tu, 1, mean)</span>
<span class="co"># # filter out genes with large estimated standard deviations</span>
<span class="co"># condSt &lt;- (DT_St_sg &lt; 0.99)</span>
<span class="co"># condTu &lt;- (DT_Tu_sg &lt; 0.99)</span>
<span class="co"># DT_Tu_m &lt;- as.numeric(apply(log2(testr.SA$ExprT), 1, mean))</span>
<span class="co"># DT_St_m &lt;- as.numeric(apply(log2(testr.TA$ExprT), 1, mean))</span>
<span class="co"># OB_St_m &lt;- OB_St_m[condSt]</span>
<span class="co"># OB_Tu_m &lt;- OB_Tu_m[condTu]</span>
<span class="co"># DT_St_m &lt;- DT_St_m[condSt]</span>
<span class="co"># DT_Tu_m &lt;- DT_Tu_m[condTu]</span>
<span class="co"># </span>
<span class="co"># # TPlot B - Mean expressions for Tumor</span>
<span class="co"># smoothScatter((DT_Tu_m + OB_Tu_m) / 2, DT_Tu_m - OB_Tu_m, </span>
<span class="co">#               ylab = &quot;Estimate - Truth&quot;, xlab = &quot;(Estimate + Truth)/2&quot;, </span>
<span class="co">#               xlim = c(2,16), ylim = c(-1.2,1.2), </span>
<span class="co">#               main = &quot;Mean expressions for Tumor&quot;, </span>
<span class="co">#               pch = 1, nrpoints = 0, col = 'yellow', </span>
<span class="co">#               colramp=colorRampPalette(c(&quot;white&quot;,&quot;yellow&quot;,&quot;yellow1&quot;,&quot;orange&quot;,&quot;orange1&quot;)))</span>
<span class="co"># tmp01 &lt;- lowess((DT_Tu_m - OB_Tu_m) ~ ((DT_Tu_m + OB_Tu_m) / 2))</span>
<span class="co"># lines(tmp01$x, tmp01$y, col=&quot;blue&quot;, lwd = 5)</span>
<span class="co"># abline(h = 0, col = 'red', lty = 2)</span>
<span class="co"># </span>
<span class="co"># # Plot C - Mean expressions for Stroma</span>
<span class="co"># smoothScatter((DT_St_m + OB_St_m) / 2, DT_St_m - OB_St_m, </span>
<span class="co">#               ylab = &quot;Estimate - Truth&quot;, xlab = &quot;(Estimate + Truth)/2&quot;, </span>
<span class="co">#               xlim = c(2,16), ylim = c(-1.2,1.2), </span>
<span class="co">#               main = &quot;Mean expressions for Stroma&quot;, pch = 1, nrpoints = 0, </span>
<span class="co">#               col = 'yellow', </span>
<span class="co">#               colramp=colorRampPalette(c(&quot;white&quot;,&quot;yellow&quot;,&quot;yellow1&quot;,&quot;orange&quot;,&quot;orange1&quot;)))</span>
<span class="co"># tmp01 &lt;- lowess((DT_St_m - OB_St_m) ~ ((DT_St_m + OB_St_m) / 2))</span>
<span class="co"># lines(tmp01$x, tmp01$y, col=&quot;blue&quot;, lwd = 5)</span>
<span class="co"># abline(h = 0, col = 'red', lty = 2)</span></code></pre></div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
